spring-kafka-2/src/main/java/com/baeldung/spring/kafka/multiplelisteners/BookEvent.java

```diff
@@ -0,0 +1,15 @@
+package com.baeldung.spring.kafka.multiplelisteners;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```

Comment:
I think @Data annotation already includes @NoArgsConstructor and @AllArgsConstructor.

Reply:
The @Data annotation includes only the  @RequiredArgsConstructor, and for this DTO, it generates only the @NoArgsConstructor. I didn't found a way that @Data to generate both  @AllArgsConstructor and @NoArgsConstructor.